enterprise/core-edge/src/main/java/org/neo4j/coreedge/raft/RaftInstance.java

```diff
@@ -226,13 +278,14 @@ protected void handleOutcome( Outcome<MEMBER> outcome ) throws RaftStorageExcept
 
             logShipping.handleCommands( outcome.getShipCommands(), leaderContext );
         }
-        else if ( oldLeader != null && oldLeader.equals( myself ) && !outcome.getLeader().equals( myself ) ) // TODO: inspect the reported issue
+        else if ( myself.equals( oldLeader ) && !myself.equals( outcome.getLeader() ) )
         {
             logShipping.stop();
         }
 
         // Update state
         state.update( outcome );
+        notifyAll();
```

Comment:
Right, that one shouldn't be protected, it's just a helper to the main synchronized handle. I'll fix and clarify it.